src/shared/components/ResourceEditor/index.tsx

```diff
@@ -84,43 +156,148 @@ const ResourceEditor: React.FunctionComponent<ResourceEditorProps> = props => {
     }
     onMetadataChange?.(checked);
   };
-  const renderCodeMirror = (value: string) => {
-    return (
-      <Spin spinning={busy}>
-        <CodeMirror
-          value={value}
-          autoCursor={false}
-          detach={false}
-          options={{
-            readOnly: !editable,
-            mode: { name: 'javascript', json: true },
-            theme: 'base16-light',
-            lineNumbers: true,
-            lineWrapping: true,
-            viewportMargin: Infinity,
-            foldGutter: true,
-            // @ts-ignore
-            foldCode: true,
-            gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter'],
-            extraKeys: {
-              'Ctrl-Q': keyFoldCode,
+  const onLinksFound = () => {
+    const elements = document.getElementsByClassName('cm-string');
+    Array.from(elements).forEach(item => {
+      const itemSpan = item as HTMLSpanElement;
+      if (isValidUrl(itemSpan.innerText.replace(/^"|"$/g, ''))) {
+        itemSpan.style.textDecoration = 'underline';
+      }
+    });
+  };
+  const onLinkClick = async (_: any, ev: MouseEvent) => {
```

Comment:
I think this function could be extracted outside and unit tested for the 3 cases. The ref may have to be passed as an arg.